### PR TITLE
Fix #156: Display x-units when available

### LINKED ISSUE #156: Show units for the x-axis
The x-axis doesn't have any units right now. We should change that.

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

**Comment by openjck** [comment]
To test this, you'll need to add a unit to a chart. Follow these steps to do that:

```
$ make shell

[docker stuff is printed]

bash-4.3$ ./manage.py shell

[python version stuff is printed]

>>> from viewer.api.models import Metric
>>> m = Metric.objects.get(pk=250)
>>> m.units = 'GPU(s)'
>>> m.save()
```

The *fxa_configured_gpu* chart will now have a unit of **GPU(s)**.

File: viewer/static/js/app/components/containers/chart-axis-container.js [hunk 1]
```diff
@@ -35,6 +35,16 @@ export default class extends React.Component {
       axis.ticks(6, d3Format.format('.0%'));
       axisElm.call(axis);
     }
+
+
+    if (props.axisType === 'x' && props.xunit) {
+      let svgElm = d3Selection.select(`.chart-${props.metricId} svg`);
```
Comment (spasovski) [inline viewer/static/js/app/components/containers/chart-axis-container.js]: we don't change this later so can be const

File: viewer/static/js/app/components/containers/chart-axis-container.js [hunk 2]
```diff
@@ -35,6 +35,16 @@ export default class extends React.Component {
...
+      svgElm.append('text')
+            .attr('class', 'label')
+            .attr('text-anchor', 'middle')
+            .attr("transform", `translate(${(props.width + 20) / 2}, ${props.height - 20})`)
```
Comment (spasovski) [inline viewer/static/js/app/components/containers/chart-axis-container.js]: `'transform'` for consistency

File: viewer/static/js/app/components/views/chart-hover.js [hunk 3]
```diff
@@ -47,15 +47,15 @@ export default class extends React.Component {
 
         // Insert the hover text for this population at this data point
         let hoverSummary = select(`.secondary-menu-content .chart-info .hover-summary[data-population="${populationName}"]`);
-        hoverSummary.html(this._getHoverString(props.metricType, d.x, d.y, proportion, populationName, props.populations[populationName].numObs));
+        hoverSummary.html(this._getHoverString(props.metricType, d.x, d.y, proportion, populationName, props.xunit, props.populations[populationName].numObs));
```
Comment (spasovski) [inline viewer/static/js/app/components/views/chart-hover.js]: Line's getting a little long I'd break up the params.

File: viewer/static/js/app/components/views/chart.js [hunk 4]
```diff
@@ -88,13 +88,18 @@ export default class extends React.Component {
...
+                height={props.size.height}
+                width={props.size.width}
+                xunit={props.xunit}
               />
               <ChartAxisContainer
                 metricId={props.metricId}
                 scale={props.yScale}
                 axisType="y"
                 refLabels={props.refLabels}
                 size={props.size.innerWidth}
```
Comment (spasovski) [inline viewer/static/js/app/components/views/chart.js]: We should't have done this...think it was me. `size` should be consistent. Just an observation unrelated to this commit.

**Comment by spasovski** [comment]
We keep using that bottom bar for more info though it's good to start thinking about how else to present all that elsewhere. This PR is fine and it makes sense within the current UI. I'll test this locally but looks good so far.

**Comment by openjck** [comment]
> We keep using that bottom bar for more info though it's good to start thinking about how else to present all that elsewhere. This PR is fine and it makes sense within the current UI. I'll test this locally but looks good so far.

Definitely. This PR does also show it below the axis, but the bottom bar definitely is getting cramped.

File: viewer/static/js/app/components/containers/chart-axis-container.js (hunk 1 above)
Comment (openjck) [inline viewer/static/js/app/components/containers/chart-axis-container.js]: Yep, works for me!

File: viewer/static/js/app/components/containers/chart-axis-container.js (hunk 2 above)
Comment (openjck) [inline viewer/static/js/app/components/containers/chart-axis-container.js]: +1

File: viewer/static/js/app/components/views/chart-hover.js (hunk 3 above)
Comment (openjck) [inline viewer/static/js/app/components/views/chart-hover.js]: Will do!

**Comment by openjck** [comment]
Updated!

File: viewer/static/js/app/components/views/chart.js (hunk 4 above)
Comment (openjck) [inline viewer/static/js/app/components/views/chart.js]: I'm not sure if I understand what you mean. What should we be doing instead?

Comment (spasovski) [inline viewer/static/js/app/components/views/chart.js]: We use 'size' to mean width (within the child component) in some cases and possibly other things (like the size object containing DOM dimensions) later depending on context. We could send all of the size object or just rename things better. Maybe I shouldn't have mentioned it as it's not specific to this commit but it seems weird now when I see the project from the outside.

**Comment by spasovski** [comment]
r+

**Pull request merged**